scrapy_statsd_extension/__init__.py

```diff
@@ -0,0 +1,41 @@
+import statsd
+from scrapy import signals
+from twisted.internet import task
+
+
+class StatsdExtension(object):
+
+    def __init__(self, crawler):
+        host = crawler.settings.get('STATSD_HOST', 'localhost')
+        port = crawler.settings.get('STATSD_PORT', 8125)
+
+        self.client = statsd.StatsClient(host, port)
+        self.stats = crawler.stats
+        self._last_stats = {}
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        ext = cls(crawler)
+
+        crawler.signals.connect(
+            ext.spider_opened, signal=signals.spider_opened)
+        crawler.signals.connect(
+            ext.spider_closed, signal=signals.spider_closed)
+
+        return ext
+
+    def spider_opened(self, spider):
+        self.task = task.LoopingCall(self.log_stats, spider)
+        self.task.start(10)
+
+    def log_stats(self, spider):
+        for key, value in self.stats.get_stats().items():
+            if isinstance(value, int) or isinstance(value, float):
+                increment_amount = value - self._last_stats.get(key, 0)
+                self.client.incr(key, increment_amount)
```

Comment:
I'd abstract `key` into a function like `self.create_stat_key` which takes `key` as an input since I think it'll probably need some modifications in the future.  It can just return `key` for now though.

Reply:
Actually, maybe it would be better to abstract the `self.client.incr` call itself into a function, rather than just the key?

Reply:
I created a create_stats_key function to format keys. Right now it just returns the key as suggested it will be easier to test. The way it works is that statsd does it's own formatting so things like:  "log_level/Error" becomes "log_level-Error"